### PR TITLE
kernel: coretemp and cpuid packages

### PR DESCRIPTION
**Pull request by pprindeville**
Add support for the Intel Coretemp (on-CPU thermal sensors) and the CPUID instruction.

### CODE REVIEW

File: package/kernel/linux/modules/hwmon.mk [hunk 1]
```diff
@@ -78,6 +78,21 @@ endef
 $(eval $(call KernelPackage,hwmon-adt7475))
 
 
+define KernelPackage/hwmon-coretemp
+  TITLE:=Pentium thermal monitoring support
+  KCONFIG:=CONFIG_SENSORS_CORETEMP
+  FILES:=$(LINUX_DIR)/drivers/hwmon/coretemp.ko
+  AUTOLOAD:=$(call AutoProbe,coretemp)
+  $(call AddDepends/hwmon)
+endef
```
Comment (hauke) [inline package/kernel/linux/modules/hwmon.mk]: This is at least activated by default in target/linux/x86/64/config-default 
Should this also be activated by default for more x86 configurations?

File: package/kernel/linux/modules/other.mk [hunk 2]
```diff
@@ -135,6 +135,18 @@ endef
...
 
+define KernelPackage/cpuid
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Intel CPUID support
+  KCONFIG:=CONFIG_X86_CPUID
+  DEPENDS:=@TARGET_x86
+  FILES:=$(LINUX_DIR)/arch/x86/kernel/cpuid.ko
+  AUTOLOAD:=$(call AutoLoad,20,cpuid)
+endef
+
```
Comment (hauke) [inline package/kernel/linux/modules/other.mk]: This is activated by default for target/linux/x86/geode/config-default
Should this also be activated by default for more x86 configurations?

**Comment by nbd168** [comment]
Yes

**Comment by nbd168** [comment]
Yes, this should be enabled in the relevant kernel configs instead of being packaged as kmod-* packages.

File: package/kernel/linux/modules/hwmon.mk (hunk 1 above)
Comment (pprindeville) [reply package/kernel/linux/modules/hwmon.mk]: > Should this also be activated by default for more x86 configurations?

@hauke We could, but then (a) `hwmon` would need to be baked into the kernel, and do we do it for all sensors, i.e. `CORETEMP`, `K10TEMP`, `K8TEMP`, `VIA_CPUTEMP`?

Comment (nbd168) [reply package/kernel/linux/modules/hwmon.mk]: Yes

File: package/kernel/linux/modules/other.mk (hunk 2 above)
Comment (pprindeville) [reply package/kernel/linux/modules/other.mk]: > Should this also be activated by default for more x86 configurations?

@Hauke it's not a big driver and more processors support it than not.

**Comment by pprindeville** [comment]
> Yes, this should be enabled in the relevant kernel configs instead of being packaged as kmod-* packages.

@nbd168 Done, see competing PR #1471.

**Pull request closed without merging**